kubernetes-deployment/massdriver.yaml

```diff
@@ -18,61 +18,98 @@ params:
   properties:
     namespace:
       title: Kubernetes Namespace
-      type: string
+      description: Application will be deployed into this namespace. If the namespace doesn't exist, it will be created.
+      $ref: https://raw.githubusercontent.com/massdriver-cloud/artifact-definitions/main/definitions/types/k8s-resource-name.json
       default: default
     image:
       title: Container Image
       type: object
+      required:
+      - repository
+      - tag
       properties:
         repository:
+          $md.immutable: true
           title: Repository
           # Container repositories can be created or imported into massdriver at: https://app.massdriver.cloud/container-repositories
           # You'll generally want to hard code your container repository in your bundle code, but it can be helpful to show it
           # in your schema as well.
           # If you need to allow an end-user developer to select the container repo, you can use our drop down widget documented here: https://docs.massdriver.cloud/bundles/custom-widgets-and-fields
-          description: "Container repository to use for this application."
+          description: Container repository to use for this application.
           type: string
-          enum:
-            - nginxdemos/hello
         tag:
-          title: Tag
-          description:
-          type: string
-          minLength: 1
+          title: Image Tag
+          $ref: https://raw.githubusercontent.com/massdriver-cloud/artifact-definitions/main/definitions/types/docker-image-tag.json
           default: latest
     resourceRequests:
       type: object
+      required:
+      - cpu
+      - memory
       properties:
         cpu:
           title: CPU
```

Comment:
Nit, thoughts?

How much CPU do you want? Ehh 100, 300 milli things? What if we called this `cores`? With a default of `0.1`-`0.5` "cores". The property name doesn't have to change, just the title / description, and adding a default.

Reply:
I like this. Changing the field title to `CPU Cores`